jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy

```diff
@@ -361,6 +361,15 @@ def step081_customize_jruby_jar(script_dir, jruby_jar, server_era) {
 }
 
 def step085_customize_hocon_settings(script_dir, settings, server_era) {
+    if ("${MAX_INSTANCES}" != "default") {
+        settings = settings ?: []
+        settings << [
+                      file: "/etc/puppetlabs/puppetserver/conf.d/puppetserver.conf",
+                      path: "jruby-puppet.maxt-active-instances",
+                      value: "${MAX_INSTANCES}"
+                    ]
+    }
+
     if (settings == null) {
```

Comment:
We'll never go down this branch now, right?

Reply:
We will, if `MAX_INSTANCES` is left "default" AND there is no `hocon_settings` block in the job definition. I think we have at least one setting in all existing jobs, but in theory we could have one without it.